tests/ut/runtime/test_device_tensor.py

```
@@ -0,0 +1,93 @@
+            DeviceTensor(0x100, [4, 0], torch.float32)
+
+    def test_negative_dim_raises(self):
+        with pytest.raises(ValueError, match="all positive"):
+            DeviceTensor(0x100, [-1, 4], torch.float32)
+
+    def test_empty_shape_raises(self):
+        with pytest.raises(ValueError, match="non-empty"):
+            DeviceTensor(0x100, [], torch.float32)
+
+    def test_wrong_dtype_type_raises(self):
+        with pytest.raises(TypeError, match="torch.dtype"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)